libnetwork/etchosts/hosts.go

```diff
@@ -0,0 +1,202 @@
+package etchosts
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+const (
+	// DefaultHostsFile is the default path to the hosts file
+	DefaultHostsFile       = "/etc/hosts"
+	hostContainersInternal = "host.containers.internal"
+	localhost              = "localhost"
+)
+
+type HostEntries []HostEntry
+
+type HostEntry struct {
+	IP    string
+	Names []string
+}
+
+// New will create a new hosts file and write this to the target file.
+// This function does not prevent any kind of concurrency problems, it is
+// the callers responsibility to avoid concurrent writes to this file.
+//
+//   - baseFile is the file where we read entries from and add entries to
+//     the target hosts file. If the name is empty it will not read any entries.
+//   - extraHosts is a slice of entries in the "hostname:ip" format. Optional.
+//   - containerIPs should contain the main container ipv4 and ipv6 if available
+//     with the container name and host name as names set. Optional.
+//   - hostContainersInternalIP is the IP for the host.containers.internal entry. Optional.
+//   - targetFile where the hosts are written to.
+//
+// The extraHosts are written first, then the hosts from the file baseFile and the
+// containerIps. The container ip entry is only added when the name was not already
+// added before.
```

Comment:
1. Not sure if such formatting is rendered properly by godoc (and go.dev website).
2. Since we have have lots of arguments, maybe move them to a structure and pass a pointer to that structure to `New`. This way, you can document the parameters in the structure, and the call will look more readable.

E.g.
```go
err := etchosts.New(&etchosts.Params{
    BaseFile: file,
    ExtraHosts: nil, // can be omitted if empty
    ContainerIPs: ips,
    HostContainerInternapIP: intIP,
    TargetFIle: tfile,
})
```